cli/flox-rust-sdk/src/providers/catalog.rs

```diff
@@ -152,11 +164,41 @@ impl CatalogClient {
                 .connect_timeout(conn_timeout)
                 .timeout(req_timeout)
                 .user_agent(format!("flox-cli/{}", &*FLOX_VERSION))
-                .default_headers(header_map)
+                .default_headers(headers)
         };
-        Self {
-            client: APIClient::new_with_client(baseurl, client.build().unwrap()),
+        APIClient::new_with_client(config.catalog_url.as_str(), client.build().unwrap())
+    }
+
+    fn build_header_map(config: &CatalogClientConfig) -> HeaderMap {
+        // let mut headers: BTreeMap<String, String> = BTreeMap::new();
+        let mut header_map = HeaderMap::new();
+
+        // Pass in a bool if we are running in CI, so requests can reflect this in the headers
+        if std::env::var("CI").is_ok() {
+            header_map.insert(
+                header::HeaderName::from_static("flox-ci"),
+                header::HeaderValue::from_static("true"),
+            );
+        };
+
+        // Authenticated requests (for custom catalogs) require a token.
+        if let Some(token) = &config.floxhub_token {
+            header_map.insert(
+                header::HeaderName::from_static("authorization"),
+                header::HeaderValue::from_str(&format!("bearer {}", token.clone())).unwrap(),
```

Comment:
nit (style, nb): no need to clone format takes token as a reference

```suggestion
                header::HeaderValue::from_str(&format!("bearer {token}")).unwrap(),
```